From 60aeaad16a6efd44513cfc0be1edaf7b1e920964 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Tue, 18 Oct 2016 04:20:50 +0000
Subject: [PATCH] OS-5731 secflags panic in shmem_lock Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Reviewed by: Robert Mustacchi
 <rm@joyent.com> Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

---
 usr/src/uts/i86pc/vm/vm_machdep.c | 3 ++-
 usr/src/uts/sun4/vm/vm_dep.c      | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/i86pc/vm/vm_machdep.c b/usr/src/uts/i86pc/vm/vm_machdep.c
index d27effaf70..0625e37bbf 100644
--- a/usr/src/uts/i86pc/vm/vm_machdep.c
+++ b/usr/src/uts/i86pc/vm/vm_machdep.c
@@ -24,7 +24,7 @@
 /*
  * Copyright (c) 2010, Intel Corporation.
  * All rights reserved.
- * Copyright (c) 2014, Joyent, Inc. All rights reserved.
+ * Copyright 2016 Joyent, Inc.
  */
 
 /* Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T */
@@ -947,6 +947,7 @@ valid_usr_range(caddr_t addr, size_t len, uint_t prot, struct as *as,
 		return (RANGE_BADADDR);
 
 	if ((addr <= (caddr_t)forbidden_null_mapping_sz) &&
+	    as->a_proc != NULL &&
 	    secflag_enabled(as->a_proc, PROC_SEC_FORBIDNULLMAP))
 		return (RANGE_BADADDR);
 
diff --git a/usr/src/uts/sun4/vm/vm_dep.c b/usr/src/uts/sun4/vm/vm_dep.c
index f80ecf7cf7..835d321c1d 100644
--- a/usr/src/uts/sun4/vm/vm_dep.c
+++ b/usr/src/uts/sun4/vm/vm_dep.c
@@ -21,6 +21,7 @@
 /*
  * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright 2016 Joyent, Inc.
  */
 
 /*
@@ -386,6 +387,7 @@ valid_usr_range(caddr_t addr, size_t len, uint_t prot, struct as *as,
 		return (RANGE_BADADDR);
 
 	if ((addr <= (caddr_t)forbidden_null_mapping_sz) &&
+	    as->a_proc != NULL &&
 	    secflag_enabled(as->a_proc, PROC_SEC_FORBIDNULLMAP))
 		return (RANGE_BADADDR);
 
-- 
2.21.0

